frontend/app/utils/date-utils.ts

```diff
@@ -69,3 +72,192 @@ export function isValidTimeZone(timeZone: string): boolean {
+ * @throws {TypeError} If the input `date` is not a valid date representation that can be parsed,
+ *   or if the `timezone` string is not a valid IANA time zone name, or if `getStartOfDayInTimezone`
+ *   or `isToday` throw an error.
+ *
+ * @example
+ * ```javascript
+ * const validDate = new Date();
+ * const invalidDateInFuture = "2078-05-20";
+ * const timezone = 'Canada/Eastern';
+ *
+ * console.log(isTodayInTimezone(timezone, validDateInPaste)); // true
+ * console.log(isTodayInTimezone(timezone, invalidDateInFuture)); // false
+ * ```
+ */
+export function isTodayInTimezone(timezone: string, date: number | string | Date): boolean {
+  return isToday(getStartOfDayInTimezone(timezone, date));
+}
+
+/**
+ * Checks if a given date is in the past or present (today) within a specified timezone.
+ *
+ * This function determines whether the provided date falls on or before the current date
+ * when considering the given timezone.  It accepts the date as a number (Unix timestamp in milliseconds),
+ * a string (parsable by the Date constructor), or a Date object.
+ *
+ * @param timeZone - The IANA time zone name (e.g., 'America/New_York', 'Europe/London').
+ * @param {number | string | Date} date - The date to be validated. Can be a Unix timestamp (milliseconds),
+ *   a date string, or a Date object.
+ * @returns {boolean} `true` if the date is in the past or today in the specified timezone, `false` otherwise.
+ * @throws {TypeError} If the input `date` is not a valid date representation that can be parsed,
+ *   or if the `timezone` string is not a valid IANA time zone name.
+ *
+ *
+ * @example
+ * ```javascript
+ * const validDateInPaste = "2023-05-20";
+ * const invalidDateInFuture = "2078-05-20";
+ * const timezone = 'Canada/Eastern';
+ * console.log(isDateInPastOrTodayInTimeZone(timezone, validDateInPaste)); // true
+ * console.log(isDateInPastOrTodayInTimeZone(timezone, invalidDateInFuture)); // false
+ * ```
+ */
+export function isDateInPastOrTodayInTimeZone(timezone: string, date: number | string | Date): boolean {
+  return isPastInTimezone(timezone, date) || isTodayInTimezone(timezone, date);
+}
+
+/**
+ * Returns the start of the day in the specified timezone.
+ *
+ * This function takes a date (or timestamp) and a timezone, and returns a new Date object
+ * representing the start of the day (midnight) in that timezone.  If no date is provided,
+ * the current date and time are used.
+ *
+ * @param {string} timezone The IANA timezone string (e.g., 'America/New_York', 'Europe/London').
+ * @param {number | string | Date} [date] Optional date or timestamp to use. If not provided, the current date and time are used.
+ *                                      Can be a number (milliseconds since epoch), a string (parsable by `new Date()`), or a Date object.
+ * @returns {Date} A new Date object representing the start of the day in the specified timezone.
```

Comment:
```suggestion
 * @param timezone - The IANA timezone string (e.g., 'America/New_York', 'Europe/London').
 * @param date - Optional date or timestamp to use. If not provided, the current date and time are used.
 *                                      Can be a number (milliseconds since epoch), a string (parsable by `new Date()`), or a Date object.
 * @returns A new Date object representing the start of the day in the specified timezone.
```

You don't have to declare the types in JSDoc when using typescript.